source/server/options_impl.cc

```diff
@@ -24,9 +24,9 @@ OptionsImpl::OptionsImpl(int argc, char** argv, const std::string& hot_restart_v
   log_levels_string += "\n[trace] and [debug] are only available on debug builds";
 
   TCLAP::CmdLine cmd("envoy", ' ', VersionInfo::version());
-  TCLAP::ValueArg<uint64_t> base_id(
+  TCLAP::ValueArg<uint32_t> base_id(
```

Comment:
Why are these necessary?

Reply:
I get this error with the pi cross-compiler. Not sure if there is some other fix, not an expert in c++... 

In file included from /ws/istio-master/src/tclap/include/tclap/Arg.h:54:0,
                 from /ws/istio-master/src/tclap/include/tclap/SwitchArg.h:30,
                 from /ws/istio-master/src/tclap/include/tclap/CmdLine.h:27,
                 from /ws/istio-master/envoy/source/server/options_impl.cc:13:
/ws/istio-master/src/tclap/include/tclap/ArgTraits.h: In instantiation of 'struct TCLAP::ArgTraits<long long unsigned int>':
/ws/istio-master/src/tclap/include/tclap/ValueArg.h:403:66:   required from 'void TCLAP::ValueArg<T>::_extractValue(const string&) [with T = long long unsigned int; std::string = std::basic_string<char>]'
/ws/istio-master/src/tclap/include/tclap/ValueArg.h:363:29:   required from 'bool TCLAP::ValueArg<T>::processArg(int*, std::vector<std::basic_string<char> >&) [with T = long long unsigned int]'
/ws/istio-master/envoy/source/server/options_impl.cc:120:1:   required from here
/ws/istio-master/src/tclap/include/tclap/ArgTraits.h:80:39: error: 'long long unsigned int' is not a class, struct, or union type
     typedef typename T::ValueCategory ValueCategory;

Reply:
Also - I don't think any of the options would need 64 bit parameters.

Reply:
That's with arm-rpi-4.9.3-linux-gnueabihf

Reply:
Might be an issue in converting from the literal to the type. I think in general, for clean 32/64-bit support, we should have some macros for writing fixed width literals, e.g. `CONST64(0)`, since we're going to hit this elswhere. I do agree that these are fine as `uint32_t` though.

Reply:
Seems specific to tclap macros - so far everything else seems to work on 32 bit arm, I'll do more testing and try to get it into a continuous build.